tpm2_pytss/crypto.py

```diff
@@ -291,3 +292,75 @@ def get_digest_size(alg):
         raise ValueError(f"unsupported digest algorithm: {alg}")
 
     return dt.digest_size
+
+
+def verify_signature_rsa(signature, key, data):
+    dt = _get_digest(signature.signature.rsapss.hash)
+    if dt is None:
+        raise ValueError(
+            f"unsupported digest algorithm: {signature.signature.rsapss.hash}"
+        )
+    if signature.sigAlg == lib.TPM2_ALG_RSASSA:
+        pad = padding.PKCS1v15()
+    elif signature.sigAlg == lib.TPM2_ALG_RSAPSS:
+        pad = padding.PSS(mgf=padding.MGF1(dt()), salt_length=dt.digest_size)
```

Comment:
TL;DR Check with `slen == hlen` if fail try `slen == PSS.MAX_LENGTH`

Oh PSS on TPMs.... So salt_len is often maximized and not equal to hash length. This has been a nightmare to deal with in tpm2-pkcs11 code. If the FIPS bit is set it's required to be the largest size allowed by FIPS which in most cases is slen == hlen. If it's not set slen == MAX. I newer specs and TPMs, hlen == slen, but older TPM's may continue the old behavior.

Reply:
I see, I added an simple trial and error solution